ivc/src/verifier.rs

```diff
@@ -0,0 +1,261 @@
+        Evaluation, PolyComm,
+    },
+    pairing_proof::{PairingProof, PairingSRS},
+    OpenProof, SRS,
+};
+use rand::thread_rng;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+
+pub type Pairing = kimchi_msm::BN254;
+/// The curve we commit into
+pub type G = kimchi_msm::BN254G1Affine;
+/// Scalar field of the curve.
+pub type Fp = kimchi_msm::Fp;
+/// The base field of the curve
+/// Used to encode the polynomial commitments
+pub type Fq = ark_bn254::Fq;
+
+pub fn verify<
+    EFqSponge: Clone + FqSponge<Fq, G, Fp>,
+    EFrSponge: FrSponge<Fp>,
+    FC: FoldingConfig<Column = GenericColumn, Curve = G, Challenge = PlonkishChallenge>,
+    const N_WIT: usize,
+    const N_REL: usize,
+    const N_DSEL: usize,
+    const N_FSEL: usize,
+    const N_ALPHAS_QUAD: usize,
+    const NPUB: usize,
+>(
+    domain: EvaluationDomains<Fp>,
+    srs: &PairingSRS<Pairing>,
+    combined_expr: &FoldingCompatibleExpr<FC>,
+    fixed_selectors: Box<[Evaluations<Fp, R2D<Fp>>; N_FSEL]>,
+    proof: &Proof<N_WIT, N_REL, N_DSEL, N_FSEL, G, PairingProof<Pairing>>,
+) -> bool {
+    assert!(N_WIT == N_REL + N_DSEL);
+
+    let Proof {
+        proof_comms,
+        proof_evals,
+        opening_proof,
+        ..
+    } = proof;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Re-evaluating public inputs
+    ////////////////////////////////////////////////////////////////////////////
+
+    let fixed_selectors_evals_d1: Box<[Evaluations<Fp, R2D<Fp>>; N_FSEL]> = fixed_selectors;
+
+    let fixed_selectors_polys: Box<[DensePolynomial<Fp>; N_FSEL]> = {
+        o1_utils::array::vec_to_boxed_array(
+            fixed_selectors_evals_d1
+                .into_par_iter()
+                .map(|evals| evals.interpolate())
+                .collect(),
+        )
+    };
+
+    let fixed_selectors_comms: Box<[PolyComm<G>; N_FSEL]> = {
+        let comm = |poly: &DensePolynomial<Fp>| srs.commit_non_hiding(poly, 1);
+        o1_utils::array::vec_to_boxed_array(
+            fixed_selectors_polys
+                .as_ref()
+                .into_par_iter()
+                .map(comm)
+                .collect(),
+        )
+    };
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Absorbing all the commitments to the columns
+    ////////////////////////////////////////////////////////////////////////////
+
+    let mut fq_sponge = EFqSponge::new(G::other_curve_sponge_params());
+
+    fixed_selectors_comms
+        .as_ref()
+        .iter()
+        .chain(&proof_comms.witness_comms)
+        .for_each(|comm| absorb_commitment(&mut fq_sponge, comm));
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Sampling scalars
+    ////////////////////////////////////////////////////////////////////////////
+
+    //let (joint_combiner, beta) = (None, Fp::zero());
+
+    // Sample α with the Fq-Sponge.
```

Comment:
I could remove these comments. I think `alpha` has to go for sure (as in we don't need to sample it). The joint combiner and beta we don't need it seems? I guess I wanted to double check.